### PR TITLE
update wasmVM to v1.4.0

### PR DESCRIPTION
**Pull request by faddat**
## Description

WasmVM has updated to a much newer version of wasmer, which should bring greatly improved performance.

This PR updates wasmvm to v1.4.0, which updates wasmer from the v2.0.x series to the v4.0.x series of wasmer.  

### Commit Message / Changelog Entry

```bash
deps: update wasmvm (credit to @misko9 for test fixes)
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

## Attribution

@misko9 made meaningful contributions to this PR that ensured that tests pass.  This is the case for all PR's linked to issue #4612 

* https://github.com/cosmos/ibc-go/pull/4306

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,9 +11,10 @@ RUN test -n "${IBC_GO_VERSION}"
 
 # Grab the static library and copy it to location that will be found by the linker flag `-lwasmvm_muslc`.
 # TODO: nice to have: a script to download version matching the wasmvm version in go.mod.
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.2.4/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
-RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep ce3d892377d2523cf563e01120cb1436f9343f80be952c93f66aa94f5737b661
-RUN cp /lib/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
+RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 2a72c7062e3c791792b3dab781c815c9a76083a7997ce6f9f2799aaf577f3c25
+RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep 8ea2e3b5fae83e671da2bb51115adc88591045953f509955ec38dc02ea5a7b94
```
Comment (chatton) [inline Dockerfile:17]: looks like we need to update the `modules/light-clients/08-wasm/Dockerfile` instead. But there's no harm actually just updating both for now.

Comment (DimitrisJim) [reply Dockerfile:17]: hopefully in the call today we'll be able to clean up the dep we currently have on 08-wasm in ibc-go. when that's done, a couple of things will be able to be removed including these changes to the Dockerfile

Comment (DimitrisJim) [reply Dockerfile:17]: we also wouldn't need to get the aarch64 lib? We should only be running actual tests on x86_64 machine.

**Comment by colin-axner** [comment]
looks like the feature branch got updated when it was bumped to v0.50 https://github.com/cosmos/ibc-go/blob/feat/wasm-clients/modules/light-clients/08-wasm/go.mod#L22, so I think we can close this pr?

**Comment by DimitrisJim** [comment]
@colin-axner we still need the bump of the wasmvm static lib. this should be easier to do in a separate PR. Closing this and will open PR bumping wasmvm. Thanks @faddat!

**Pull request closed without merging**